packages/providers/src/provider.ts

```diff
@@ -205,28 +207,75 @@ export type ProviderCallParams = {
   utxoValidation?: boolean;
 };
 
+/**
+ * URL - Consensus Params mapping.
+ */
+type ConsensusParamsCache = Record<string, ConsensusParameters>;
+
 /**
  * A provider for connecting to a node
  */
 export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
+  consensusParamsCache: ConsensusParamsCache = {};
```

Comment:
instead of caching `consensusParams` , we can anticipate the needings and cache whole `chainInfo` , as we have it in hands.

we can also create same approach and for `nodeInfo`, from `provider.getNodeInfo()`

Reply:
Great point!